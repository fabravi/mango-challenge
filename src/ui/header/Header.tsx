"use client";

import Link from "next/link";
import Image from "next/image";
import styles from "./header.module.scss";
import { usePathname } from "next/navigation";

export default function Navigation() {
  const pathname = usePathname();
  const isActive = (href: string) => pathname === href;

  return (
    <header className={styles.header}>
      <Link href="/">
        <Image
          src="/mango.svg"
          alt="Mango Logo"
          width={200}
          height={50}
          priority
        />
      </Link>

      <nav className={styles.menu}>
        <ul>
          <li>
            <Link
              href="/exercise1"
              className={isActive("/exercise1") ? styles.active : ""}
            >
              Exercise 1
            </Link>
          </li>
          <li>
            <Link
              href="/exercise2"
              className={isActive("/exercise2") ? styles.active : ""}
            >
              Exercise 2
            </Link>
          </li>
        </ul>
      </nav>
    </header>
  );
}
