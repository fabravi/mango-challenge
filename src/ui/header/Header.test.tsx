import { render } from "@testing-library/react";
import Header from "./Header";

describe("Header", () => {
  test("should render correctly", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  test("should render links", () => {
    const { container } = render(<Header />);
    const links = container.querySelectorAll("a");
    expect(links).toHaveLength(3);
  });
});
