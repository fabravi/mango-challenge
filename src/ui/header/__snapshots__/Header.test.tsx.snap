// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div>
  <header
    class="header"
  >
    <a
      href="/"
    >
      <img
        alt="Mango Logo"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="50"
        src="/mango.svg"
        style="color: transparent;"
        width="200"
      />
    </a>
    <nav
      class="menu"
    >
      <ul>
        <li>
          <a
            class=""
            href="/exercise1"
          >
            Exercise 1
          </a>
        </li>
        <li>
          <a
            class=""
            href="/exercise2"
          >
            Exercise 2
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
